### DISCUSSION TITLE
Can we use nodemailer in Hono

### DISCUSSION THREAD

**Discussion by Kuldeep12mohan** [honojs/hono#3096] [Q&A] [answered]
I am curious about how can i send mails using hono as http layer, i'm facing error like nodeJS compatibility issue

**Comment by Meess** [comment]
Could you elaborate a bit more on what issue you're encountering. E.g. what runtime are you using, what is the error you're getting, the smallest code example that you can make to reproduce the issue? Otherwise it very difficult to determine what goes wrong

**Reply by Kuldeep12mohan** [reply]
Actually i am using cloudflare workers for deployment purpose it could be possible that, that might be causing problem here is the piece of code 
`import { Hono } from "hono";
import { patientRouter } from "./routes/patient";
import { cors } from "hono/cors";
import nodemailer from "nodemailer";
import { gmailPass } from "./constant";

const app = new Hono<{
  Bindings: {
    DATABASE_URL: string;
  };
}>();

const transporter = nodemailer.createTransport({
  service: 'gmail',
  auth: {
    user: 'mohankuldeep1234@gmail.com',
    pass: gmailPass  // Ensure gmailPass is handled securely
  }
});

app.use("/*", cors());

app.post("/send-verification", async (c) => {
  try {
    const { email } = await c.req.json();
    
    const mailOptions = {
      from: 'mohankuldeep1234@gmail.com',
      to: email,
      subject: "Verification Email",
      text: "Your verification message goes here."
    };

    const info = await transporter.sendMail(mailOptions);
    console.log("Email sent:", info);

    c.status(200);
    return c.json({
      message: "Email sent successfully"
    });
  } catch (error) {
    console.error("Error sending email:", error);
    c.status(500);
    return c.json({
      message: "Something went wrong"
    });
  }
});
export default app`


I'm facing error like Node js compatibility

**Reply by Meess** [reply, marked as answer]
nodemailer is not compatible with cloudflare's workers (i.e. workerd runtime), see the issue in the nodemailer repo: https://github.com/nodemailer/nodemailer/issues/1621#issuecomment-2143562757

So this is not a hono issue, but a nodemailer <-> cloudflare worker runtime issue. So to answer your original question: **Can we use nodemailer in Hono?**
- Yes, you can use nodemailer with hono, but not with cloudflare worker runtime as it's not supported by the nodemailer package

**Reply by Kuldeep12mohan** [reply]
thanks! I got the answer

**Reply by Meess** [reply]
Great, if you can mark the ticket as answered, then other know it's already resolved and don't have to dive in :).
